### ISSUE TITLE
Node triggers turns to OFF at midnight, while schedule runs till 00:30...

### ISSUE THREAD

**Issue by spudje** [niklaswall/node-red-contrib-light-scheduler#64]
I don't get a schedule to work that runs from a certain time in one day, until some time the next day, e.g. Fri 9:00 till Sa 00:30. I've dragged a schedule to the next day, so it extends on Friday till 0:00 and then continues on Saturday until 00:30. However when running in just schedule mode, the node triggers OFF at 0:00., not at 0:30.

**Comment by wizbowes** [comment]
Also suffering from this.  I'm sure it didn't use to do this.

Version: 0.0.16
Node-Red Version: 1.1.3

**Comment by JanJoh** [comment]
Yeah, I found this when searching for info why Light Scheduler sends state at midnight regardless of if "only on state change" is set. 

I noticed this in a flow where i have most of the day (but not all) as on, and want to leverade sun position.

**Comment by niklaswall** [comment]
I have not tested this yet, but sounds like a real problem.

It looks like changes to the graphical scheduling component has introduced a problem. It should not be possible to add a schedule that starts one day and ends another day (and it's still not possible when creating a new schedule). It is however possible to resize (drag-and drop) the end of a schedule to another day.

The way it originally worked was that you create a schedule until midnight, and if you want it to stay on after midnight, you create a new schedule the day after. This is most likely also a work-around that solves the problem. i.e **never create a schedule that extends past midnight**.

I have set this up to reproduce the problem, but would be great if anyone else can verify that this workaround solves the problem.

//Niklas

**Comment by JanJoh** [comment]
In. my case i did not do that. I created "per day".  And a new "on" is sent at midnight.

My current workaround is a "RBE" node on the output. Dirty. I know 🙂
________________________________
Från: Niklas Wall <notifications@github.com>
Skickat: den 25 november 2020 11:57
Till: niklaswall/node-red-contrib-light-scheduler <node-red-contrib-light-scheduler@noreply.github.com>
Kopia: Jan Johansson <j2@mupp.net>; Comment <comment@noreply.github.com>
Ämne: Re: [niklaswall/node-red-contrib-light-scheduler] Node triggers turns to OFF at midnight, while schedule runs till 00:30... (#64)


I have not tested this yet, but sounds like a real problem.

It looks like changes to the graphical scheduling component has introduced a problem. It should not be possible to add a schedule that starts one day and ends another day (and it's still not possible when creating a new schedule). It is however possible to resize (drag-and drop) the end of a schedule to another day.

The way it originally worked was that you create a schedule until midnight, and if you want it to stay on after midnight, you create a new schedule the day after. This is most likely also a work-around that solves the problem. i.e never create a schedule that extends past midnight.

I have set this up to reproduce the problem, but would be great if anyone else can verify that this workaround solves the problem.

//Niklas

**Comment by wizbowes** [comment]
Any update on this?  I have a schedule on starting at 6pm finishing at midnight, and then another one starting at midnight to 6am - the idea being the light will be on from 6pm to 6am.  Always turns off at midnight each night.

**Comment by navarro152** [comment]
Good to know that I am not the only one having this issue. Same here, but with my heathers, I have some of them to turn off at 1 am, but sometimes if we go to bed earlier I take the temperature lower before. So the option of setting another trigger after midnight it might work as a workaround, but a few days it might be off.
But, besides, this, thanks for the node, it is very good :)

**Comment by niklaswall** [comment]
Just released version 0.0.17 that has some fixes related to this issue. It does however require that you edit the node and save it again for the changes to take affect.

Please let me know if this fixes your issues! (I'm running some tests to try to reproduce this issue meanwhile)

**Comment by navarro152** [comment]
Hello, and thanks for the work on the add-on. 

I tried yesterday, now it does not let me create a schedule that past midnight, at least now I will not be expecting to have my schedule to go after that, and in any case, I will have to add another one.

**Comment by niklaswall** [comment]
Yes, it was never intended to be possible to create events that cross over midnight. This was a bug introduced when updating the editor.

If you have schedules that is back-to-back with each other that should not cause any events to be sent at the time of the stop/start between schedules.
